includes/frontend/class-wcal-checkout-process.php

```diff
@@ -214,10 +225,18 @@ public function wcal_updated_recovered_cart( $cart_id, $order_id, $wcal_check_em
 
 				// Add Order Note.
 				$order->add_order_note( __( 'This order was abandoned & subsequently recovered.', 'woocommerce-abandoned-cart' ) );
-				delete_post_meta( $order_id, 'wcal_abandoned_cart_id' );
-				delete_post_meta( $order_id, 'wcal_recover_order_placed' );
-				delete_post_meta( $order_id, 'wcal_recover_order_placed_sent_id' );
-				delete_post_meta( $order_id, 'wcal_recovered_email_sent' );
+				if ( is_hpos_enabled() ) {
```

Comment:
The condition check for hpos_enabled or no is not needed in this file at all. Please remove the checks at all the places and always use the $order object to add/update/delete meta data. You can refer to [this](https://github.com/TycheSoftwares/woocommerce-abandon-cart-pro/blob/master/includes/frontend/wcap_on_placed_order.php) file in the Pro version if needed.